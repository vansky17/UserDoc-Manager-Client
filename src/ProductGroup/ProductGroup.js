import React from 'react'
import '../App/App.css'
import Doc from '../Doc/Doc'
import UserDocsContext from "../UserDocsContext";
/* import uuid from "uuid";  */
class ProductGroup extends React.Component {

  static defaultProps = {
    match: {
      params: {}
    }
  }

  static contextType = UserDocsContext;
  render () {
    const { name, id, docs } = this.props
    return (
      <>
        <section id={id}>
        {/* <header> */}
        <h3>{name}</h3>
        {/* </header> */}
        <table>
          <tbody>
          <tr>
            <th>Document</th><th>Part Number</th><th>Doc Version</th><th>Format</th><th>Release Date</th>
          </tr> 
{/*           <Doc key={uuid.v4()}/>
          <Doc key={uuid.v4()}/>
          <Doc key={uuid.v4()}/> */}
          { docs.map(doc =>
            <Doc
              key={doc.id}
              id={doc.id}
              name={doc.name}
              partnum={doc.partnum}
              vernum={doc.vernum}
              formattype={doc.formattype}
              reldate={doc.reldate}
              descr={doc.descr}
              author={doc.author}
              productid={doc.productid}
              path={doc.path}
            >    
            </Doc>
          ) }
      </tbody>
        </table>
      </section>
    </>
    )
  }
}
ProductGroup.defaultProps = {
  docs: [],
};
export default ProductGroup
