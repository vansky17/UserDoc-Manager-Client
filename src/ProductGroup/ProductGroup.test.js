import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ProductGroup from './ProductGroup'

describe(`ProductGroup component`, () => {
  const props = {
    id: 'a',
    name: 'test',
    docs: [
      {
        "id": 24,
        "name": "practice for mockup2",
        "partnum": "34563456",
        "vernum": 1,
        "formattype": "mp4",
        "reldate": "2020-02-12T21:27:48.510Z",
        "author": "John Doe",
        "productid": 6,
        "descr": "No description",
        "path": "https://userdocsmanager.s3.us-east-2.amazonaws.com/notes.png"
    },
    {
        "id": 25,
        "name": "Video",
        "partnum": "127",
        "vernum": 1,
        "formattype": "mp4",
        "reldate": "2020-02-12T21:31:02.972Z",
        "author": "IS",
        "productid": 6,
        "descr": "No description",
        "path": "https://userdocsmanager.s3.us-east-2.amazonaws.com/small.mp4"
    },
    {
        "id": 26,
        "name": "JS-playground 2",
        "partnum": "123.4567",
        "vernum": 1,
        "formattype": "DOCX",
        "reldate": "2020-02-12T22:28:23.585Z",
        "author": "IS",
        "productid": 6,
        "descr": "No description",
        "path": "https://userdocsmanager.s3.us-east-2.amazonaws.com/clean_form_bg.jpg"
    }
    ]
  }

  it('renders a .ProductGroup by default', () => {
    const wrapper = shallow(<ProductGroup />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the ProductGroup given props', () => {
    const wrapper = shallow(<ProductGroup {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
