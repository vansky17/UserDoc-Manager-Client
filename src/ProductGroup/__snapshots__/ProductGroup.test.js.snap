// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGroup component renders a .ProductGroup by default 1`] = `
<Fragment>
  <section>
    <h3 />
    <table>
      <tbody>
        <tr>
          <th>
            Document
          </th>
          <th>
            Part Number
          </th>
          <th>
            Doc Version
          </th>
          <th>
            Format
          </th>
          <th>
            Release Date
          </th>
        </tr>
      </tbody>
    </table>
  </section>
</Fragment>
`;

exports[`ProductGroup component renders the ProductGroup given props 1`] = `
<Fragment>
  <section
    id="a"
  >
    <h3>
      test
    </h3>
    <table>
      <tbody>
        <tr>
          <th>
            Document
          </th>
          <th>
            Part Number
          </th>
          <th>
            Doc Version
          </th>
          <th>
            Format
          </th>
          <th>
            Release Date
          </th>
        </tr>
        <Doc
          author="John Doe"
          descr="No description"
          formattype="mp4"
          id={24}
          key="24"
          name="practice for mockup2"
          partnum="34563456"
          path="https://userdocsmanager.s3.us-east-2.amazonaws.com/notes.png"
          productid={6}
          reldate="2020-02-12T21:27:48.510Z"
          vernum={1}
        />
        <Doc
          author="IS"
          descr="No description"
          formattype="mp4"
          id={25}
          key="25"
          name="Video"
          partnum="127"
          path="https://userdocsmanager.s3.us-east-2.amazonaws.com/small.mp4"
          productid={6}
          reldate="2020-02-12T21:31:02.972Z"
          vernum={1}
        />
        <Doc
          author="IS"
          descr="No description"
          formattype="DOCX"
          id={26}
          key="26"
          name="JS-playground 2"
          partnum="123.4567"
          path="https://userdocsmanager.s3.us-east-2.amazonaws.com/clean_form_bg.jpg"
          productid={6}
          reldate="2020-02-12T22:28:23.585Z"
          vernum={1}
        />
      </tbody>
    </table>
  </section>
</Fragment>
`;
